### DISCUSSION TITLE
RQをどうするか

### DISCUSSION THREAD

**Discussion by ritsuxis** [ritsuxis/SoftwareEcosystem-Group1#5] [Ideas]
- whitelist/blacklist, master/slaveをふさわしくない単語の対象とする
  - 他の単語を探しても結局この2つが上位になる気がする

**Comment by kuuchan-code** [comment]
whitelist/blacklist, master/slaveは好ましい単語へどのくらいの割合だけ置き換えられているか？今後どのように置き換えが推移すると予想されるか？

**Reply by kuuchan-code** [reply]
方法として適当に選んだGitHub上のリポジトリにおいてデフォルトブランチ名がmasterとそれ以外のもののである場合を数える。もしある時間のデフォルトブランチ名も調べることができるのならば、時間と置き換えられた割合とで回帰することで、今後どのように置き換えが推移するか予想ができるかもしれない。

**Comment by ritsuxis** [comment]
リサーチ
RQ1. GitHubのスター数上位100個のリポジトリのうちデフォルトブランチの名前がmasterになっているものはどれぐらいあるか
RQ2. main/それ以外になっている場合は何時置き換えられたのか、それに対する議論にはどのようなものがあるか
RQ3. 

今後ホットになりそうな差別ワードとか…？

**Reply by kuuchan-code** [reply]
RQ2の調べ方がまだわかっていません

**Reply by ritsuxis** [reply]
RQ2はGitHubのGraphQLのAPIを使用してリポジトリに上がっているIssueとPRから置き換え時期を図ろうとしていました
そのための（解析していない）データを今しがたこのリポジトリに上げたので、検索条件の改善点や他にほしいデータがあれば、言ってもらえれば対応できます

**Comment by ritsuxis** [comment]
2020/10/1にGitHubのデフォルトのブランチ名がmainになった
https://github.blog/changelog/2020-10-01-the-default-branch-for-newly-created-repositories-is-now-main/

**Comment by celorind** [comment]
Linuxは2020/7/10

https://japan.zdnet.com/article/35156658/

**Comment by ritsuxis** [comment]
授業のフォーラムの方に編集用のスライドをアップしました

**Comment by kuuchan-code** [comment]
ealps授業ページの記載に従って、「なぜそれを知りたいか，それがわかるとどう実務者に役立つか」も議論したほうが良さそうですね。

**Reply by kuuchan-code** [reply]
RQ1に関して（案）
なぜそれを知りたいのか：どの程度のGitHub / Git ユーザがmainといった好ましい単語へ置き換えるべきだと判断しているかの指標になり得るから
それがわかるとどう実務者に役立つか：その数値を知ってもらうことで、デフォルトブランチ名決定時のきっかけになるかもしれない

**Reply by kuuchan-code** [reply]
ですが差別問題は多数決で良い悪いを決定できるものではないかもしれないとも思っています

**Reply by kuuchan-code** [reply]
ある人が差別と感じた時点で差別なので、そう感じる人を少しでも減らすような選択をすべきかもしれない

**Reply by ritsuxis** [reply]
今回のmaster/mainに限らず、自分があまり興味のない差別表現に対して世間ではどこまで配慮すれば現時点で無難なのかを知りたい気持ちはありますね

**Reply by ritsuxis** [reply]
あと企業とかのリポジトリは燃えた時により周りがうるさいので、個人よりも積極的な取り組みはしていると思っています
これは体感なので今回でこの傾向を実際に確認できればなと
